tests/logreq_lock_desired.test/runit

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+bash -n "$0" | exit 1
+
+#export debug=1
+[[ $debug == "1" ]] && set -x
+export stopfile=./stopfile.txt
+
+. ${TESTSROOTDIR}/tools/cluster_utils.sh
+. ${TESTSROOTDIR}/tools/runit_common.sh
+
+function create_table
+{
+    $CDB2SQL_EXE $CDB2_OPTIONS $DBNAME default "create table if not exists x(a int)"
+}
+
+function insert_loop
+{
+    while [[ ! -f $stopfile ]]; do
+        $CDB2SQL_EXE $CDB2_OPTIONS $DBNAME default "insert into x select * from generate_series(1, 100000)" >/dev/null 2>&1
+    done
+}
+
+function downgrade_and_wait
+{
+    master=$($CDB2SQL_EXE -tabs $CDB2_OPTIONS $DBNAME default "select host from comdb2_cluster where is_master='Y' limit 1")
+    $CDB2SQL_EXE $CDB2_OPTIONS $DBNAME --host $master "exec procedure sys.cmd.send('downgrade')" >/dev/null 2>&1
+}
+
+function run_test
+{
+    rm -f $stopfile >/dev/null 2>&1
+
+    create_table
+    insert_loop &
+    insert_pid=$!
+
+    sleep 10
+    foundtrace=0
+
+    while [[ "$foundtrace" == "0" ]]; do
+
+        replicant=""
+        while [[ -z "$replicant" ]]; do
+            replicant=$($CDB2SQL_EXE -tabs $CDB2_OPTIONS $DBNAME default "select host from comdb2_cluster where is_master='N' limit 1")
+            if [[ -z "$replicant" ]]; then
+                sleep 5
+            fi
+        done
+        replicant=$($CDB2SQL_EXE -tabs $CDB2_OPTIONS $DBNAME default "select host from comdb2_cluster where is_master='N' limit 1")
+
+        echo "Bringing $replicant down for 120 seconds"
+        kill_restart_node $replicant 120 0
+
+        echo "Sleeping for 40 seconds for recovery"
+        sleep 40
+
```

Comment:
Not super familiar with this code, but is there something that we can do here to force the replicant to sleep in the log_get loop here so that we can avoid running this test in a loop? Would just running some inserts do the trick?

Reply:
Yeah, sure - lemme rework a bit

Reply:
Modified the order a bit & increased the sleep-time- looks to be reliably reproducing without looping.